pkg/api/server/v1alpha2/ordering.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Tekton Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var allowedOrderByFields = []string{"created_time", "updated_time"}
+
+// orderBy validates and returns a string formatted suitably for
+// a sql order by clause.
+func orderBy(fields string) (string, error) {
+	if strings.TrimSpace(fields) == "" {
+		return "", nil
+	}
+
+	var orderBy []string
+	for _, field := range strings.Split(fields, ",") {
+		ob, err := normalizeOrderByField(field)
+		if err != nil {
+			return "", err
+		}
+		orderBy = append(orderBy, ob)
+	}
+
+	return strings.Join(orderBy, ","), nil
+}
+
+// normalizeOrderByField takes a field string, validating and formatting
+// it for use in a sql query. An error is returned if the format of the string
+// doesn't match either "field_name" or "field_name direction".
+func normalizeOrderByField(field string) (string, error) {
+	field = strings.TrimSpace(field)
+	f := strings.Fields(field)
+	fieldName := ""
+	direction := ""
+	switch len(f) {
+	case 1:
+		fieldName = f[0]
+	case 2:
+		fieldName = f[0]
+		direction = f[1]
+	default:
+		errmsg := fmt.Sprintf("invalid order_by %q", field)
+		return "", status.Error(codes.InvalidArgument, errmsg)
+	}
+
+	fieldName = strings.TrimSpace(fieldName)
+
+	if !isAllowedField(fieldName) {
+		errmsg := fmt.Sprintf("order by %s not supported", fieldName)
+		return "", status.Error(codes.InvalidArgument, errmsg)
+	}
+
+	if direction == "" {
+		return fieldName, nil
+	} else {
+		direction = strings.TrimSpace(direction)
+		return orderByDirection(fieldName, direction)
+	}
+}
+
+func isAllowedField(name string) bool {
+	for i := range allowedOrderByFields {
+		if name == allowedOrderByFields[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func orderByDirection(field string, direction string) (string, error) {
+	sortOrder := strings.ToLower(strings.TrimSpace(direction))
```

Comment:
The `TrimSpace`s in this func is redundant with the trimming already done in `normalizeOrderByField`. Can we consolidate these?